src/main/resources/com/toptrumps/online/scripts/models/player.js

```diff
@@ -0,0 +1,88 @@
+const PlayerModel = (($) => {
+    let players = {};
+    let playersList = {};
+
+    const init = playersData => {
+        playersData.forEach(player => {
+            players[player.id] = player;
+            playersList[player.id] = { id: player.id, name: player.name };
+        });
+    };
+
+    const getPlayers = () => {
+        return players;
+    };
+
+    const getPlayersList = () => {
+        return playersList;
+    };
+
+    const getTopCard = playerID => {
+        return players[playerID].deck.shift();
+    };
+
+    const getAiPlayersTopCards = () => {
+        const playersCards = {};
+
+        $.each(players, (i, player) => {
+            if (player.id === 0) return;
+
+            playersCards[player.id] = getTopCard(player.id);
+        });
+
+        return playersCards;
+    };
+
+    const prepareDataset = cardsOnTable => {
+        let dataset = [];
+
+        cardsOnTable.forEach(data => {
+            dataset.push({
+                id: data.playerID,
+                isAIPlayer: data.playerID !== 0,
+                name: players[data.playerID].name,
+                deckCount: players[data.playerID].deck.length + 1,
+                topCard: data.card
+            });
+        });
+
+        return dataset;
+    };
+
+    const getPlayersCardCount = () => {
```

Comment:
For every getter operation, you're having to traverse through a ```players``` array. Why not define a ```players``` variable inside your controller and extract all info you need at the beginning of each round -- so you can loop through it once.